lib/CLI/Parser.hs

```diff
@@ -0,0 +1,550 @@
+
+viewOptions :: Parser ViewOptions
+viewOptions = do
+  ViewOptions
+    <$> argument readPath ( mconcat
+          [ metavar "PATH"
+          , help "The path to either a directory of entries, or a single entry"
+          ])
+    <*> optional
+          ( argument readFieldKey $ mconcat
+              [ metavar "FIELDNAME"
+              , help "The optional field name to fetch the contents of"
+              ]
+          )
+
+createOptions :: Parser CreateOptions
+createOptions =
+  CreateOptions
+    <$> argument readEntryPath ( mconcat
+          [ metavar "ENTRYPATH"
+          , help "The path to insert the new entry into, this must not already be a directory or an entry unless `-f` is specified"
+          ])
+    <*> switch ( mconcat
+          [ long "edit"
+          , short 'e'
+          , help $ unlines
+              [ "Open an editor to configure the fields and tags for this entry."
+              , "By default, `vi` will be used."
+              , "This can be set using the `$EDITOR` environment variable."
+              ]
+          ])
+    <*> switch ( mconcat
+          [ long "force"
+          , short 'f'
+          , help "If a directory or entry already exists under the specified path, delete it and insert this entry instead"
+          ])
+    <*> many ( option readEntryTag $ mconcat
+          [ long "tag"
+          , metavar "TAG"
+          , help "Tag to add to this new entry, this may be specified multiple times"
+          ])
+    <*> many ( option readFieldInfo $ mconcat
+          [ long "field"
+          , metavar "NAME=CONTENT"
+          , help "A field to insert into the new entry, with the format 'fieldname=fieldcontents', this may be specified multiple times"
+          ])
+    <*> many ( option readFieldInfo $ mconcat
+          [ long "privatefield"
+          , metavar "NAME=CONTENT"
+          , help $ unlines
+              [ "Same as `--field`, but the field will be marked as private."
+              , "Private fields can only be viewed with 'coffer view',"
+              , "and will be hidden when using other commands."
+              ]
+          ])
+
+setFieldOptions :: Parser SetFieldOptions
+setFieldOptions =
+  SetFieldOptions
+    <$> argument readEntryPath ( mconcat
+          [ metavar "ENTRYPATH"
+          , help "The path to set the field value on, this must already exist as an entry"
+          ])
+    <*> argument readFieldKey ( mconcat
+          [ metavar "FIELDNAME"
+          , help "The name of the field to set"
+          ])
+    <*> optional (argument str $ mconcat
+          [ metavar "FIELDCONTENTS"
+          , help "The contents to insert into the field. Required when creating a new field, optional otherwise"
+          ])
+    <*> optional (option readFieldVisibility $ mconcat
+          [ long "visibility"
+          , short 'V'
+          , metavar "VISIBILITY"
+          , help "Whether to mark this field as 'public' or 'private'"
+          ])
```

Comment:
Oh good point, I hadn't considered that. I guess the implementation is also wrong, because the spec says "public" should be the default. I'll change this.